src/pages/Governance/utils.tsx

```diff
@@ -30,62 +31,74 @@ function canBeResolvedEarly(proposal: Proposal): boolean {
   return false;
 }
 
-/* TODO - calculate/fetch proposal status */
 // replicate on-chain logic get_proposal_state()
 // https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/voting.move
-export function getProposalState(proposal: Proposal): ProposalState {
+export function getProposalState(proposal: Proposal): ProposalVotingState {
   if (isVotingClosed(proposal)) {
     let yesVotes = parseInt(proposal.yes_votes);
     let noVotes = parseInt(proposal.no_votes);
     let minVoteThreshold = proposal.min_vote_threshold;
 
-    if (yesVotes > noVotes && yesVotes + noVotes >= minVoteThreshold) {
-      return ProposalState.SUCCEEDED;
+    if (yesVotes <= noVotes && yesVotes + noVotes >= minVoteThreshold) {
+      return ProposalVotingState.REJECTED; // more "no" votes
+    } else if (yesVotes > noVotes && yesVotes + noVotes >= minVoteThreshold) {
+      return ProposalVotingState.PASSED; // more "yes" votes
     } else {
-      return ProposalState.FAILED;
+      return ProposalVotingState.FAILED; // not enough votes
     }
   } else {
-    return ProposalState.PENDING;
+    return ProposalVotingState.PENDING;
   }
 }
 
-export function getStatusColor(proposalState: ProposalState) {
+export function getVotingStatusColor(proposalState: ProposalVotingState) {
   switch (proposalState) {
-    case ProposalState.SUCCEEDED:
+    case ProposalVotingState.PASSED:
       return primaryColor;
-    case ProposalState.PENDING:
+    case ProposalVotingState.PENDING:
       return warningColor;
-    case ProposalState.FAILED:
+    case ProposalVotingState.FAILED:
+      return negativeColor;
+    case ProposalVotingState.REJECTED:
       return negativeColor;
```

Comment:
Add a default case with assertNever for exhaustiveness checking. https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#union-exhaustiveness-checking